src/Models/HydrostaticFreeSurfaceModels/HydrostaticFreeSurfaceModels.jl

```diff
@@ -0,0 +1,40 @@
+module HydrostaticFreeSurfaceModels
+
+using KernelAbstractions: @index, @kernel, Event, MultiEvent
+using KernelAbstractions.Extras.LoopInfo: @unroll
+
+using Oceananigans.Utils: launch!
+
+import Oceananigans: fields
+
+#####
+##### HydrostaticFreeSurfaceModel definition
+#####
+
+include("compute_w_from_continuity.jl")
+include("hydrostatic_free_surface_tendency_fields.jl")
+include("hydrostatic_free_surface_model.jl")
+include("show_hydrostatic_free_surface_model.jl")
+include("set_hydrostatic_free_surface_model.jl")
+
+#####
+##### Time-stepping HydrostaticFreeSurfaceModels
+#####
+
+"""
+    fields(model::HydrostaticFreeSurfaceModel)
+
+Returns a flattened `NamedTuple` of the fields in `model.velocities` and `model.tracers`.
+"""
+fields(model::HydrostaticFreeSurfaceModel) = merge((u = model.velocities.u,
+                                                    v = model.velocities.v,
+                                                    η = model.free_surface.η),
+                                                    model.tracers)
```

Comment:
I'm guessing this is really `prognostic_fields` since it doesn't include `model.velocities.w`?

Reply:
Yeah, and this is also true for `IncompressibleModel` (doesn't include pressures, diffusivities, etc). We should change the name of this function to `prognostic_fields` (but I think this should be done in a separate PR).

Reply:
We could follow the `ShallowWaterModel` example and name these `model.solution`.  

Incidently, is the rationale for not putting `w` and pressure hear to avoid storage?